Installation issues

## Installation Instructions
I think the installation instructions for contributors are incomplete. 

At [this link](https://toqito.readthedocs.io/en/latest/getting_started.html#contributing), contributors have a link for how to clone a GitHub repo etc. But an extra line is missing for installing an editable version, i.e., using `pip install -e .` in the top-level directory.  

I can make a PR for this if it is needed. 

## Installation Error
Edit : I managed to solve the python version error. But I am not sure how to install an editable version of `toqito`. 

If I use `pip install -e .`, there's an error `error: can not execute 'setup.py' since setuptools is not available in the build environment` even though the latest version of `setuptools` is involved. 

```
Installing collected packages: numpy, cvxopt, scipy, picos, scs, qdldl, ecos, osqp, cvxpy, toqito
  Running setup.py develop for toqito
    error: subprocess-exited-with-error
    
    × python setup.py develop did not run successfully.
    │ exit code: 1
    ╰─> [1 lines of output]
        ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× python setup.py develop did not run successfully.
│ exit code: 1
╰─> [1 lines of output]
    ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.

```

Great. Happy to incorporate the `setuptools` change in your diff. 

As for the test failure, I've opened a separate issue for that here. Thanks for catching that!

> Ok. I think I managed to fix the `setuptools is not available in the build environment` error. Not sure why it works though because I have a surface-level understanding of how `pyproject.toml` works. [See this comment for more](https://github.com/python-poetry/poetry/issues/4242#issuecomment-917655570).
> 
> When I used `pip install -e .`, `setuptools` was being uninstalled during the process even though it was specified in `pyrpoject.toml`.
> 
> If I make the following changes in lines 63 & 69, I don't get the installation error.
> 
> ![image](https://user-images.githubusercontent.com/66048318/240732301-fa6e5fb5-868c-4093-aaf2-ca37b29e2852.png)
> 
> Edit:
> 
> There's also [one failing test](https://github.com/vprusso/toqito/blob/master/tests/test_matrix_props/test_sk_norm.py#L48) when I run `pytest --cov-report term-missing --cov=toqito tests/` locally.
> 
> ```
> n = 2, a = 0.5
> 
>     @pytest.mark.parametrize("n, a", [(2, 0.5), (2, -0.5), (3, 0.5), (3, -0.5)])
>     def test_s1_norm_werner(n, a):
>         """Test S(1) norm of a Werner state."""
>         rho = werner(n, a)
>         # See:
>         # N. Johnston.
>         # Norms and Cones in the Theory of Quantum Entanglement.
>         # PhD thesis (arXiv:1207.1479)
>         # Proposition 5.2.10 and Table 5.1
>         expected = (1 + abs(min(0, a))) / (n * (n - a))
>     
>         lower_bound, upper_bound = sk_operator_norm(rho, k=1)
> >       np.testing.assert_equal(np.allclose(lower_bound, expected), True)
> E       AssertionError: 
> E       Items are not equal:
> E        ACTUAL: False
> E        DESIRED: True
> 
> tests/test_matrix_props/test_sk_norm.py:60: AssertionError
> ```
https://github.com/vprusso/toqito/issues/151

--- a/.github/CONTRIBUTING.md
+++ b/.github/CONTRIBUTING.md
@@ -10,6 +10,8 @@ describes how to merge code changes into this `toqito`.
 *    On your local machine,
      [clone](https://help.github.com/articles/cloning-a-repository/) your fork of
      the repository.
+*    To install an editable version on your local machine, run `pip install -e .` in
+     the top-level directory of the cloned repository.
 
 ## Making Changes
 
@@ -26,7 +28,10 @@ describes how to merge code changes into this `toqito`.
      automatically.  Each commit added to the PR will be validated for
      mergability, compilation and test suite compliance; the results of these tests
      will be visible on the PR page.
-*    If you're providing a new feature, you must add test cases and documentation.
+*    If you're providing a new feature, you must add test cases and documentation. We use `sphinx`
+     to build the documentation. To build the documentation locally via `make html` in the
+     `toqito/docs` directory, make sure `sphinx` and `sphinx-rtd-theme` are installed.
+     For more information, visit [sphinx documentation](https://docs.readthedocs.io/en/stable/intro/getting-started-with-sphinx.html) & [sphinx-rtd-theme documentation](https://sphinx-rtd-theme.readthedocs.io/en/stable/installing.html)
 *    When the code is ready to go, make sure you run the test suite using pytest.
 *    When you're ready to be considered for merging, check the "Ready to go"
      box on the PR page to let the `toqito` devs know that the changes are complete.

--- a/docs/getting_started.rst
+++ b/docs/getting_started.rst
@@ -4,9 +4,11 @@ Getting started
 Installing
 ^^^^^^^^^^
 
-1. Ensure you have Python 3.9 or greater installed on your machine.
+1. Ensure you have Python 3.9 or greater installed on your machine or in 
+a virtual environment (`pyenv <https://github.com/pyenv/pyenv>`_, `pyenv tutorial <https://realpython.com/intro-to-pyenv/>`_). 
 
 2. Consider using a `virtual environment <https://packaging.python.org/guides/installing-using-pip-and-virtualenv/>`_.
+You can also use :code:`pyenv` with :code:`virtualenv` `to manage different Python versions <https://github.com/pyenv/pyenv-virtualenv>`_. 
 
 3. The preferred way to install the :code:`toqito` package is via :code:`pip`.
 
@@ -40,9 +42,12 @@ Python module installed and built for your machine.
 Testing
 ^^^^^^^
 
-The :code:`pytest` module is used for testing. In order to run and :code:`pytest`, you will need to ensure it is
-installed on your machine. Consult the `pytest <https://docs.pytest.org/en/latest/>`_ website for more information. To
-run the suite of tests for :code:`toqito`, run the following command in the root directory of this project:
+The :code:`pytest` module is used for testing and :code:`pytest-cov` is used to generate
+coverage reports. In order to run and :code:`pytest`, you will need to ensure it is
+installed on your machine along with :code:`pytest-cov`. Consult the `pytest <https://docs.pytest.org/en/latest/>`_ 
+and `pytest-cov <https://pytest-cov.readthedocs.io/en/latest/>`_ websites for more
+information. To run the suite of tests for :code:`toqito`, run the following command
+in the root directory of this project:
 
 .. code-block:: bash
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -61,7 +61,7 @@ mypy = "^0.750.0"
 coverage = "^4.5"
 coveralls = "^1.9"
 setuptools = ">65.5.1"
-sphinx = "3.1.2"
+sphinx = "6.2.1"
 sphinx_rtd_theme = "*"
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -60,11 +60,11 @@ pytest-cov = "*"
 mypy = "^0.750.0"
 coverage = "^4.5"
 coveralls = "^1.9"
-setuptools = "<=57.5.0"
+setuptools = ">65.5.1"
 sphinx = "3.1.2"
 sphinx_rtd_theme = "*"
 
 
 [build-system]
-requires = ["poetry>=0.12"]
+requires = ["setuptools","poetry>=0.12"]
 build-backend = "poetry.masonry.api"

